tests/integration/ObsDetails.test.js

```diff
@@ -0,0 +1,70 @@
+import { screen } from "@testing-library/react-native";
+import ObsDetails from "components/ObsDetails/ObsDetails";
+import initI18next from "i18n/initI18next";
+import inatjs from "inaturalistjs";
+import React from "react";
+
+import factory, { makeResponse } from "../factory";
+import { renderAppWithComponent } from "../helpers/render";
+
+const mockComment = factory( "LocalComment" );
+const mockObservation = factory( "LocalObservation", {
+  comments: [mockComment]
+} );
+const mockUpdate = factory( "RemoteUpdate", {
+  resource_uuid: mockObservation.uuid,
+  comment_id: mockComment.id
+} );
+
+// Mock api call to observations
+jest.mock( "inaturalistjs" );
+inatjs.observations.update.mockResolvedValue( makeResponse( [mockUpdate] ) );
+
+jest.mock( "@react-navigation/native", () => {
+  const actualNav = jest.requireActual( "@react-navigation/native" );
+  return {
+    ...actualNav,
+    addEventListener: () => {},
+    useNavigation: () => ( {
+      navigate: jest.fn(),
+      setOptions: jest.fn()
+    } ),
+    useRoute: () => ( {
+      params: {
+        uuid: mockObservation.uuid
+      }
+    } )
+  };
+} );
+
+describe( "ObsDetails", () => {
+  beforeAll( async () => {
+    await initI18next();
+
+    // Write local observation to Realm
+    await global.realm.write( () => {
+      global.realm.create( "Observation", mockObservation );
+    } );
+  } );
+
+  afterEach( () => {
+    jest.clearAllMocks();
+  } );
+
+  describe( "when there is an observation with an update", () => {
```

Comment:
This is more like "when there is an observation where we don't know if the user has viewed comments". If you want to test the case when there is definitely a new comment, you would `expect( observation.viewed_comments ).toBe( false )`, right?